controllers/complianceeventsapi/server.go

```diff
@@ -446,9 +503,145 @@ func parseQueryArgs(queryArgs url.Values, isCSV bool) (*queryOptions, error) {
+		return parsed, nil
+	}
+
+	// If there is any unAuthorized Cluster, it should return ErrNotAuthorized
+	if len(unAuthorizedClusters) > 0 {
+		log.V(2).Info("need GET access", "managedClusters", strings.Join(unAuthorizedClusters, ", "))
+
+		return nil, fmt.Errorf("need GET access to these managedClusters: %s, error: %w",
+			strings.Join(unAuthorizedClusters, ", "), ErrNotAuthorized)
+	}
+
+	if len(authorizedClusterNames) == 0 && len(authorizedClusterIds) == 0 {
+		log.V(2).Info("The service account does not have GET authorization for all")
+
+		return nil, ErrNotAuthorized
+	}
+
+	parsed.Filters["clusters.name"] = authorizedClusterNames
+	parsed.Filters["clusters.cluster_id"] = authorizedClusterIds
+
 	return parsed, nil
 }
 
+// Return authorized cluster names or, If the url query doesn't have specific clusters, it verifies
+// all managedCluster Authorizations and it returns true if the user has all authorization for manageClusters
+// getAuthorizedClusterNames returns also unauthorized managedClusters name list to give information to users
+func getAuthorizedClusters(ctx context.Context, db *sql.DB, parsed queryOptions,
+	userConfig *rest.Config, c client.Client,
+) (authorizedClusterIds, authorizedClusterNames, unAuthorizedClusters []string,
+	includeAllManagedClusters bool, err error,
+) {
+	// Get all managedCluster rules
+	allRules, err := getManagedClusterRules(userConfig, nil)
+	if err != nil {
+		return authorizedClusterIds, authorizedClusterNames, unAuthorizedClusters, false, err
+	}
+
+	clusterIDs, okClusterIds := parsed.Filters["clusters.cluster_id"]
+	if okClusterIds {
+		// Convert id to name
+		for _, id := range clusterIDs {
+			clusterName, err := getClusterNameFromID(ctx, db, id)
+			if err != nil {
+				log.Error(err, "Failed to get cluster name from cluster id :"+id)
+
+				// This leads to send forbidden error to the user
+				unAuthorizedClusters = append(unAuthorizedClusters, id)
+
+				continue
+			}
+
+			if getAccessByClusterName(allRules, clusterName) {
+				authorizedClusterIds = append(authorizedClusterIds, id)
+			} else {
+				unAuthorizedClusters = append(unAuthorizedClusters, id)
+			}
+		}
+	}
+
+	parsedClusterNames, okClusterNames := parsed.Filters["clusters.name"]
+	if okClusterNames {
+		for _, clusterName := range parsedClusterNames {
+			if getAccessByClusterName(allRules, clusterName) {
+				authorizedClusterNames = append(authorizedClusterNames, clusterName)
+			} else {
+				unAuthorizedClusters = append(unAuthorizedClusters, clusterName)
+			}
+		}
+	}
+
+	// There is no cluster.id or clusterName arg from the url query
+	// In other words, the user requests all
+	if !okClusterIds && !okClusterNames {
+		startRules, ok := allRules["*"]
+		if ok && slices.Contains(startRules, "get") || slices.Contains(startRules, "*") {
+			return authorizedClusterIds, authorizedClusterNames, unAuthorizedClusters, true, nil
+		}
+
+		managedClusterNames, err := getAllManagedClusterNames(ctx, c)
```

Comment:
If the user doesn't have `get` access on `*` managed clusters, then just set `parsed.Filters["clusters.name"]` to the managed clusters the user does have explicit `get` access to in `allRules`.

Reply:
This block is when the user doesn't have parsed.Filters["clusters.name"]

Reply:
Is` GET "/api/v1/compliance-events` for all managedcluster ? I thought we needed to verify all `managedcluster` auths. For example: `managedclusters`: m1, m2, m3
but the user has only m1 "GET". Then I thought I should return UnAuthorized